runelite-api/src/main/java/net/runelite/api/Experience.java

```diff
@@ -173,4 +176,22 @@ public static int getCombatLevel(int attackLevel, int strengthLevel,
 	{
 		return (int) getCombatLevelPrecise(attackLevel, strengthLevel, defenceLevel, hitpointsLevel, magicLevel, rangeLevel, prayerLevel);
 	}
+
+	/**
+	 * Calculate the percentages of each level compared to the difference of the goal exp and the start exp
+	 *
+	 * @param startXp Starting exp
+	 * @param goalXp The goal exp
+	 */
+	public static List<Double> levelPercentagesBetweenExperience(int startXp, int goalXp)
+	{
+		final int startLevel = Experience.getLevelForXp(startXp);
+		final int endLevel = Experience.getLevelForXp(goalXp);
+
+		return IntStream.range(startLevel + 1, endLevel)
```

Comment:
You should figure something else out so this isn't making a stream every render frame.

Reply:
I can either memoize the function in the experience class or it can be left to the caller to memoize it on it's own. I'll probably memoize it on the experience level later today, or does someone know a reason to have it memoize on the callers level?

Reply:
@raiyni It get's cached now